src/main/java/uk/gov/hmcts/reform/iacaseaccessapi/infrastructure/service/CcdSupplementaryDetailsSearchService.java

```diff
@@ -63,14 +73,31 @@ public CcdSupplementaryDetailsSearchService(SystemTokenGenerator systemTokenGene
             throw new IdentityManagerResponseException(e.getMessage(), e);
         }
 
-        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
-        TermsQueryBuilder termQueryBuilder = QueryBuilders.termsQuery("reference", ccdCaseNumberList);
-        searchSourceBuilder.size(100);
-        searchSourceBuilder.from(0);
-        searchSourceBuilder.sort("created_date", SortOrder.DESC);
-        searchSourceBuilder.query(termQueryBuilder);
+        List<SupplementaryInfo> allResults = new ArrayList<>();
+        final Collection<List<String>> chunkedCcdCaseList = getChunkedCcdCaseList(ccdCaseNumberList);
+        final int proposedPoolSize = (int) Math.ceil((double) ccdCaseNumberList.size() / maxRecords);
+        ExecutorService executorService = Executors.newFixedThreadPool(proposedPoolSize);
+
+        for (List<String> splitCcdCaseList : chunkedCcdCaseList) {
+
+            CompletableFuture<List<SupplementaryInfo>> completableFuture = CompletableFuture.supplyAsync(
+                () -> {
+                    SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+                    TermsQueryBuilder termQueryBuilder = QueryBuilders.termsQuery("reference", splitCcdCaseList);
+                    searchSourceBuilder.size(100);
+                    searchSourceBuilder.from(0);
+                    searchSourceBuilder.sort("created_date", SortOrder.DESC);
+                    searchSourceBuilder.query(termQueryBuilder);
+
+                    return search(userToken, s2sToken, searchSourceBuilder.toString());
+                },
+                executorService
+            );
+
+            allResults.addAll(completableFuture.get());
```

Comment:
You need to have a list of List<CompletableFuture<List<SupplementaryInfo>>> allResults to keep non-blocking reference.
Iterate on CompletableFuture outside on main loop (`for (List<String> splitCcdCaseList : chunkedCcdCaseList)`) so you take advantage of immediate initiation for the all futures upfront
the `get()` method triggered on the future will block and wait for the future completion
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/CompletableFuture.html#get()